lld/tools/lld/lld.cpp

```diff
@@ -136,10 +137,22 @@ static Flavor parseFlavor(std::vector<const char *> &v) {
 // and we use it to detect whether we are running tests or not.
 static bool canExitEarly() { return StringRef(getenv("LLD_IN_TEST")) != "1"; }
 
+// This function is called on startup. We need this for LTO since
+// LTO calls LLVM functions to compile bitcode files to native code.
+// Technically this can be delayed until we read bitcode files, but
+// we don't bother to do lazily because the initialization is fast.
+static void initLLVM() {
+  InitializeAllTargets();
+  InitializeAllTargetMCs();
+  InitializeAllAsmPrinters();
+  InitializeAllAsmParsers();
+}
+
```

Comment:
Wouldn't this affect the linker for other platforms, COFF, for example, by calling these functions twice, first here and then again in `LinkerDriver::link`

Reply:
Good point. I'll fix that too.

Reply:
When I checked the other targets, only coff has these calls -- I've removed them.